addons/web_editor/static/src/js/backend/linkDialogCommand.js

```diff
@@ -0,0 +1,56 @@
+/** @odoo-module **/
+
+import Wysiwyg from 'web_editor.wysiwyg'
+import { registry } from '@web/core/registry'
+import { HotkeyCommandItem } from '@web/core/commands/default_providers'
+
+let lastWysiwygAndRange;
+
+const commandProviderRegistry = registry.category("command_provider");
+commandProviderRegistry.add("link dialog", {
+    async provide() {
+        let wysiwyg;
+        let range;
+
+        if (lastWysiwygAndRange) {
+            [wysiwyg, range] = lastWysiwygAndRange;
+        } else {
+            wysiwyg = [...Wysiwyg.activeWysiwygs].find((wysiwyg) => {
+                return wysiwyg.isSelectionInEditable();
+            });
+            if (wysiwyg) {
+                const selection = wysiwyg && wysiwyg.odooEditor.document.getSelection();
+                range = selection && selection.rangeCount && selection.getRangeAt(0);
+            }
+            if (range) {
+                lastWysiwygAndRange = [wysiwyg, range];
+            }
+        }
+
+        if (range) {
+            let link = wysiwyg.getInSelection('a');
+            const label = !link ? 'Create link' : 'Edit link';
+
+            return [
+                {
+                    Component: HotkeyCommandItem,
+                    action: () => {
+                        const selection = wysiwyg.odooEditor.document.getSelection();
+                        selection.removeAllRanges();
+                        selection.addRange(range);
```

Comment:
yes, we need to use set the range in the editor because the next instructions will need that selection. 
`range` is the range before the command palette open of the focused editor.